plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/MockStorage.java

```diff
@@ -236,12 +242,16 @@ public boolean isOpen() {
         return null;
     }
 
+    private final Set<BlobInfo> simulated410s = new HashSet<>();
```

Comment:
do we need to make this concurrency-safe?

Reply:
I don't think we have to, there's no tests that would require that as far as I can tell. Made it thread-safe regardless though because the rest of this class is thread-safe as well.